heron/healthmgr/src/java/com/twitter/heron/healthmgr/sensors/MetricsCacheMetricsProvider.java

```diff
@@ -37,23 +38,28 @@
 import com.twitter.heron.proto.tmaster.TopologyMaster.MetricResponse.IndividualMetric;
 import com.twitter.heron.proto.tmaster.TopologyMaster.MetricResponse.IndividualMetric.IntervalValue;
 import com.twitter.heron.proto.tmaster.TopologyMaster.MetricResponse.TaskMetric;
+import com.twitter.heron.proto.tmaster.TopologyMaster.MetricsCacheLocation;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
 import com.twitter.heron.spi.utils.NetworkUtils;
 
-import static com.twitter.heron.healthmgr.HealthManager.CONF_METRICS_SOURCE_URL;
+import static com.twitter.heron.healthmgr.HealthManager.CONF_TOPOLOGY_NAME;
 
 public class MetricsCacheMetricsProvider implements MetricsProvider {
-  private static final String PATH_STATS = "/stats";
+  private static final String PATH_STATS = "stats";
   private static final Logger LOG = Logger.getLogger(MetricsCacheMetricsProvider.class.getName());
-  private HttpURLConnection con;
+  private SchedulerStateManagerAdaptor stateManagerAdaptor;
+  private String topologyName;
 
   private Clock clock = new Clock();
+  private String metricsCacheLocation;
 
   @Inject
-  public MetricsCacheMetricsProvider(@Named(CONF_METRICS_SOURCE_URL) String metricsCacheURL) {
-    LOG.info("Metrics will be provided by MetricsCache at :" + metricsCacheURL);
+  public MetricsCacheMetricsProvider(SchedulerStateManagerAdaptor stateManagerAdaptor,
```

Comment:
The buildCommonConfigModule() is associated with the injector inside HealthMgr. How does Guice link that inject to here.